managed/src/main/java/com/yugabyte/yw/controllers/EncryptionAtRestController.java

```diff
@@ -42,6 +48,32 @@
 
   @Inject Commissioner commissioner;
 
+  @Inject CloudAPI.Factory cloudAPIFactory;
+
+  private void validateKMSProviderConfigFormData(ObjectNode formData, String keyProvider) {
+    if (keyProvider.toUpperCase().equals(KeyProvider.AWS.toString())
+        && (formData.get("AWS_ACCESS_KEY_ID") != null
+            || formData.get("AWS_SECRET_ACCESS_KEY") != null)) {
+      CloudAPI cloudAPI = cloudAPIFactory.get(KeyProvider.AWS.toString().toLowerCase());
+      Map<String, String> config = new HashMap<String, String>();
+      config.put("AWS_ACCESS_KEY_ID", formData.get("AWS_ACCESS_KEY_ID").textValue());
+      config.put("AWS_SECRET_ACCESS_KEY", formData.get("AWS_SECRET_ACCESS_KEY").textValue());
+      if (cloudAPI != null
+          && !cloudAPI.isValidCreds(config, formData.get("AWS_REGION").textValue())) {
+        throw new YWServiceException(BAD_REQUEST, "Invalid AWS Credentials.");
+      }
+    }
+    if (keyProvider.toUpperCase().equals(KeyProvider.SMARTKEY.toString())) {
+      final Set<String> apiUrl =
+          new HashSet<>(
+              Arrays.asList("api.amer.smartkey.io", "api.eu.smartkey.io", "api.uk.smartkey.io"));
```

Comment:
Would be good for this list to be a static class field or even an enum as we don't expect it to ever change and re-building the list/set over and over is unnecessary work

Reply:
yeah make sense. Declared static class field var. Thanks!